lib/connections/service.go

```diff
@@ -265,8 +265,10 @@ next:
 		// keep up with config changes to the rate and whether or not LAN
 		// connections are limited.
 		isLAN := s.isLAN(c.RemoteAddr())
-		wr := s.limiter.newWriteLimiter(c, isLAN)
-		rd := s.limiter.newReadLimiter(c, isLAN)
+		s.limiter.mu.Lock()
```

Comment:
This is unusual, we do don't want to take locks on mutexes inside other objects and then call their `...Locked` functions. Wrap in a method.

`rd, wr := s.limiter.getLimiters(remoteID, c, isLAN)`